packages/icon-button/src/mwc-icon-button-base.ts

```diff
@@ -25,15 +28,88 @@ export class IconButtonBase extends LitElement {
 
   @property({type: String}) label = '';
 
+  @query('button') buttonElement!: HTMLElement;
+
+  @queryAsync('mwc-ripple') ripple!: Promise<Ripple|null>;
+
+  @internalProperty() protected shouldRenderRipple = false;
+
+  protected rippleHandlers: RippleHandlers = new RippleHandlers(() => {
+    this.shouldRenderRipple = true;
+    return this.ripple;
+  });
+
+  protected renderRipple() {
+    return html`${
+        this.shouldRenderRipple ? html`<mwc-ripple .disabled="${
+                                      this.disabled}" unbounded></mwc-ripple>` :
+                                  ''}`;
+  }
+
+  focus() {
+    const buttonElement = this.buttonElement;
+    if (buttonElement) {
+      this.rippleHandlers.startFocus();
+      buttonElement.focus();
+    }
+  }
+
+  blur() {
+    const buttonElement = this.buttonElement;
+    if (buttonElement) {
+      this.rippleHandlers.endFocus();
+      buttonElement.blur();
+    }
+  }
+
   /** @soyCompatible */
   protected render() {
     return html`<button
-    .ripple="${ripple()}"
-    class="mdc-icon-button"
-    aria-label="${this.label || this.icon}"
-    ?disabled="${this.disabled}">
+        class="mdc-icon-button"
+        aria-label="${this.label || this.icon}"
+        ?disabled="${this.disabled}"
+        @focus="${this.handleRippleFocus}"
+        @blur="${this.handleRippleBlur}"
+        @mousedown="${this.handleRippleActivate}"
+        @mouseenter="${this.handleRippleMouseEnter}"
+        @mouseleave="${this.handleRippleMouseLeave}"
+        @touchstart="${this.handleRippleActivate}">
+      ${this.renderRipple()}
     <i class="material-icons">${this.icon}</i>
     <slot></slot>
   </button>`;
   }
+
+  @eventOptions({passive: true})
+  private handleRippleActivate(evt?: Event) {
+    const onUp = () => {
+      window.removeEventListener('mouseup', onUp);
+      window.removeEventListener('touchend', onUp);
```

Comment:
the touch listeners can be moved back into the template as they will always be retargetted to the button unlike the mouseup